File: 킬링캠프/1주 차 - 완전탐색, 백트래킹, 이분탐색/practice/Binary Search/b. 재귀 풀이/Solution.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        return binarySearch(nums, target, left, right);
    }

    private int binarySearch(int[] nums, int target, int left, int right) {
        if (left > right) {
            return -1;
        }

        int mid = (right - left) / 2 + left;
        if (nums[mid] == target) {
            return mid;
        }

        if (nums[mid] < target) {
            return binarySearch(nums, target, mid + 1, right);
        } else {
            return binarySearch(nums, target, left, mid - 1);
        }
    }
}

File: 킬링캠프/1주 차 - 완전탐색, 백트래킹, 이분탐색/practice/Binary Search/b. 재귀 풀이/solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        # 메서드
        def binarySearch(left, right):
            if left > right:
                return -1
            
            mid = left + ((right - left) // 2)
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
                
            return binarySearch(left, right)
            
        # 메인 로직
        return binarySearch(0, len(nums) - 1)
